File: src/com/bulletphysics/util/ObjectStackList.java
package com.bulletphysics.util;

/**
 * Stack-based object pool for arbitrary objects, returning not supported.
 *
 * @author jezek2
 */
public class ObjectStackList<T> extends StackList<T> {

    private final Class<T> cls;

    public ObjectStackList(Class<T> cls) {
        super();
        this.cls = cls;
    }

    @Override
    protected T create() {
        try {
            return cls.newInstance();
        } catch (InstantiationException | IllegalAccessException e) {
            throw new IllegalStateException(e);
        }
    }
}

File: src/com/bulletphysics/util/ObjectStackList.kt
package com.bulletphysics.util

/**
 * Stack-based object pool for arbitrary objects, returning not supported.
 *
 * @author jezek2
 */
class ObjectStackList<T>(private val cls: Class<T>) : StackList<T>() {
    override fun create(): T {
        try {
            return cls.newInstance()
        } catch (e: InstantiationException) {
            throw IllegalStateException(e)
        } catch (e: IllegalAccessException) {
            throw IllegalStateException(e)
        }
    }
}
